### PR TITLE
fix updating cluster_alias

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-3793](https://issues.redhat.com/browse/COST-3793)

## Description

This change will check if an entry exists for a provider in the reporting_ocp_clusters table before creating one, if the cluster_alias is different we will also update that.

## Testing

1. Checkout Branch
2. Restart Koku
3. create/load test data
4. update a source/provider name
5. see the cluster alias is update correctly in reporting_ocp_clusters table.

## Notes

...

### CODE REVIEW

File: koku/masu/database/ocp_report_db_accessor.py [hunk 1]
```diff
@@ -1001,18 +1001,27 @@ def populate_openshift_cluster_information_tables(self, provider, cluster_id, cl
     def populate_cluster_table(self, provider, cluster_id, cluster_alias):
         """Get or create an entry in the OCP cluster table."""
         with schema_context(self.schema):
-            cluster, created = OCPCluster.objects.get_or_create(
-                cluster_id=cluster_id, cluster_alias=cluster_alias, provider_id=provider.uuid
-            )
+            cluster = OCPCluster.objects.filter(provider_id=provider.uuid).first()
+            msg = f"Fetching entry in reporting_ocp_clusters for provider: {provider.uuid}"
```
Comment (maskarb) [inline koku/masu/database/ocp_report_db_accessor.py:1005]: This log message should come first:
```suggestion
            LOG.info(log_json(msg="fetching entry in reporting_ocp_cluster", provider_uuid=provider.uuid))
            cluster = OCPCluster.objects.filter(provider_id=provider.uuid).first()
            msg = "fetched entry in reporting_ocp_cluster"
```

**Review by maskarb — COMMENTED** [review]
Do we need to fix any messed up entries in the database? Or have we already taken care of those?

Some logging nitpicks:

File: koku/masu/database/ocp_report_db_accessor.py [hunk 2]
```diff
@@ -1001,18 +1001,27 @@ def populate_openshift_cluster_information_tables(self, provider, cluster_id, cl
...
-                msg=f"created entry in reporting_ocp_clusters: {created}",
-                cluster_id=cluster_id,
-                cluster_alias=cluster_alias,
-                provider=provider,
+            LOG.info(
+                log_json(
+                    msg=msg,
+                    cluster_id=cluster_id,
+                    cluster_alias=cluster_alias,
+                    provider=provider,
```
[the remark this answers is not in the document]
Comment (lcouzens) [reply koku/masu/database/ocp_report_db_accessor.py:1022]: I never changed this but its a fair point, not a clue either :D

**Review by maskarb — APPROVED** [review]

**Pull request merged**